### DISCUSSION TITLE
Training the spacy3 sm ner model for ".Net Developer" as person (Dot Problem)

### DISCUSSION THREAD

**Discussion by rammohanrrr** [explosion/spaCy#7901] [Help: Best practices]
I have a training data to train NER pipe with custom label as ROLE.
I have trained using that data with ROLE label.

Whereas, My trained model not able to identify ".Net Developer" which I have given in training data.
Can you suggest me how to train it.
Thanks in advance

Example:
Training Data:
("Client: Apple Inc. , Role: .Net Developer", {"entities":[(27,41,"ROLE")]})  

When I give same text to predict, output is here (ent.text, ent.label_)
Output:    Developer, ROLE
Expected:    .Net Developer, ROLE

**Comment by polm** [comment]
Sometimes statistical models just make mistakes, especially if an example is different from others. 

In this case you have several options:

**Add more training examples with the thing you want to improve on.** This one is generally effective and pretty straightforward. 

**Modify the Tokenizer.** I suspect `.Net` is becoming two tokens. You can [add a tokenizer exception](https://spacy.io/usage/linguistic-features#special-cases) to treat it as one token, which should make it easier for the model to learn, since it won't have to distinguish "the period in .Net" from "the period in general". 

You can use this code to do that:

```
import spacy
from spacy.symbols import ORTH

nlp = spacy.load("en_core_web_sm")
special_case = [{ORTH: ".Net"}]
nlp.tokenizer.add_special_case(".Net", special_case)
text = "Client: Apple Inc. , Role: .Net Developer"
print(list(nlp(text)))
```

**Use rule-based entity labelling.** Since ".Net Developer" should always be labelled as a role, you can use the [rule-based matching](https://spacy.io/usage/rule-based-matching) to pick it up. Add an EntityRuler after your NER component and allow it to overwrite entities and you can pick up edge cases like this.

**Comment by rammohanrrr** [comment]
Answer is best suited to me, and this is what I'm searching...

Thanks @polm for giving good information, Great project
